### PR TITLE
FEATURE: More descriptive personal courses page

### LINKED ISSUE #4: FEATURE: More descriptive personal courses page
### Have you updated the title?

- [X] I have updated the issue's title.

### As a...

user

### I can...

view all courses

### So that...

know what courses i have

### Discussion

Please make the "All courses" display page more descriptive. Currently it only shows course ID, enroll code, & personal schedule ID. Please modify this table to display the personal schedule name, course name, and quarter, as well as any other information that may be useful.

### Acceptance Criteria

- [ ] The "All courses" table shows more useful/descriptive information
- [ ] I can view each course's name, schedule name, and quarter from the "All courses" table

### Implementation Todos

Frontend:
- [ ] Add columns to the PS Course table for course name, schedule name, and quarter.
- [ ] Format these fields correctly (i.e. quarter should be displayed in QYY format, not YYYYQ format)

Testing:
- [ ] Front end tests pass and there is adequate coverage
- [ ] Back end tests pass and there is adequate coverage

### PR DESCRIPTION
**Pull request by Torry051**
Add Course Name, Schedule Name, and Quarter for Course Page.

course

https://proj-courses-torry051-dev.dokku-08.cs.ucsb.edu

Close #4

### CODE REVIEW

**Comment by tallyhawley** [comment]
I would love to be able to access this information without having to manually enter it on the CourseForm.

**Review by tallyhawley — CHANGES_REQUESTED** [review]
I would love to be able to access this information without having to manually enter it on the CourseForm.
Please take a look at how they access course information in the Personal Schedule Details page. (Also, please format quarters in QYY format, not YYYYQ. There is a helper function to do this.)

**Comment by Torry051** [comment]
Solve all issues and waiting for code review.

image

My dokku App linked to another branch now. I will update dokku deployment latter.

File: .DS_Store
Comment (tallyhawley) [inline .DS_Store:1]: remove this

File: frontend/package-lock.json
Comment (tallyhawley) [inline frontend/package-lock.json:1]: remove these changes from your PR

File: frontend/src/tests/components/Courses/CourseForm.test.js [hunk 1]
```diff
@@ -112,3 +123,64 @@ describe("CourseForm tests", () => {
     await waitFor(() => expect(mockedNavigate).toHaveBeenCalledWith(-1));
   });
 });
+
```
Comment (tallyhawley) [inline frontend/src/tests/components/Courses/CourseForm.test.js]: dead code/commented out? delete this

File: frontend/src/tests/pages/Courses/CoursesCreatePage.test.js [hunk 2]
```diff
@@ -61,8 +61,11 @@ describe("CoursesCreatePage tests", () => {
     const courses = [
       {
         id: "17",
+        // courseName: "CMPSC",
```
Comment (tallyhawley) [inline frontend/src/tests/pages/Courses/CoursesCreatePage.test.js]: just delete this if it's going to be commented out anyway

File: frontend/src/tests/pages/Courses/CoursesCreatePage.test.js [hunk 3]
```diff
@@ -81,12 +84,20 @@ describe("CoursesCreatePage tests", () => {
...
+    // const courseNameField = screen.getByTestId("CourseForm-courseName");
+    // const schduleNameField = screen.getByTestId("CourseForm-schduleName");
+    // const quarterField = screen.getByTestId("CourseForm-quarter");
     const psIdField = document.querySelector("#CourseForm-psId");
     const submitButton = screen.getByTestId("CourseForm-submit");
 
     fireEvent.change(psIdField, { target: { value: 13 } });
     localStorage.setItem("CourseForm-psId", "13");
     fireEvent.change(enrollCdField, { target: { value: "08250" } });
+    // fireEvent.change(courseNameField, { target: { value: "CMPSC" } });
```
Comment (tallyhawley) [inline frontend/src/tests/pages/Courses/CoursesCreatePage.test.js]: same here

**Review by tallyhawley — CHANGES_REQUESTED** [review]
some small changes, should be really quick! & please update your dokku so i can test this out :)

**Comment by Torry051** [comment]
https://proj-courses-torry051-dev.dokku-08.cs.ucsb.edu

deployment update and files removed.

**Comment by tallyhawley** [comment]
Getting 404 error?
image

**Comment by Torry051** [comment]
> Getting 404 error? image


> Getting 404 error? image

This error exists because we delete the personal schedule but not delete the course. And my backend code need to use  the psid to get all other information list in the course page. 

So, I still need to show the course with extra description when the ps has been deleted?

**Comment by Torry051** [comment]
> > Getting 404 error? image
> 
> I think that error is because the Personal Schedule of the course has been deleted, but the course is still there.
> 
> > Getting 404 error? image
> 
> This error exists because we delete the personal schedule but not delete the course. And my backend code need to use the psid to get all other information list in the course page.

So, I still need to show the course with extra description when the ps has been deleted?

**Comment by Torry051** [comment]
When the schedule are deleted, the courses not be deleted will still show in index page, without any backend error. 

Here is the new deployment.

https://proj-courses-torry051-dev.dokku-08.cs.ucsb.edu


image

image

**Review by tallyhawley — APPROVED** [review]
LGTM

**Review by anirudhiyer1 — APPROVED** [review]
LGTM

**Pull request merged**